src/VisualStudio/Core/Def/Implementation/Preview/FileChange.cs

```diff
@@ -79,13 +80,18 @@ private ChangeList ComputeChildren(TextDocument left, TextDocument right, Cancel
 
             diffService = diffService ?? diffSelector.DefaultTextDifferencingService;
 
-            var diff = ComputeDiffSpans(oldText, newText, diffService, cancellationToken);
-            if (!diff.Differences.Any())
+            var diff = ComputeDiffSpans(diffService, left, right, cancellationToken);
+            if (diff.Differences.Count == 0)
             {
-                // There are no changes. Just show the whole document.
-                return GetEntireDocumentAsSpanChange(left);
+                // There are no changes.
+                return ChangeList.Empty;
```

Comment:
@heejaechang this will mean that a blank preview dialog pane will show up for such documents. I am not sure what case do we ever have a FileChange created when a document hasn't changed, probably for code actions where a fix to a diagnostic involves no change to current document but a change somewhere else in the project. If so, we may want to retain the original behavior OR not have a FileChange node for this file at all. We should make sure that we don't end up showing a tree node, such that clicking on it shows a blank pane underneath.